tsconfig.json

```
@@ -1,21 +1,95 @@
+        // "tsBuildInfoFile": "./.tsbuildinfo",              /* Specify the path to .tsbuildinfo incremental compilation file. */
+        // "disableSourceOfProjectReferenceRedirect": true,  /* Disable preferring source files instead of declaration files when referencing composite projects. */
+        // "disableSolutionSearching": true,                 /* Opt a project out of multi-project reference checking when editing. */
+        // "disableReferencedProjectLoad": true,             /* Reduce the number of projects loaded automatically by TypeScript. */
+        /* Language and Environment */
+        "target": "ES2020", /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
+        // "lib": [],                                        /* Specify a set of bundled library declaration files that describe the target runtime environment. */
+        // "jsx": "preserve",                                /* Specify what JSX code is generated. */
+        "experimentalDecorators": true, /* Enable experimental support for TC39 stage 2 draft decorators. */
+        "emitDecoratorMetadata": true, /* Emit design-type metadata for decorated declarations in source files. */
+        // "jsxFactory": "",                                 /* Specify the JSX factory function used when targeting React JSX emit, e.g. 'React.createElement' or 'h'. */
+        // "jsxFragmentFactory": "",                         /* Specify the JSX Fragment reference used for fragments when targeting React JSX emit e.g. 'React.Fragment' or 'Fragment'. */
+        // "jsxImportSource": "",                            /* Specify module specifier used to import the JSX factory functions when using 'jsx: react-jsx*'. */
+        // "reactNamespace": "",                             /* Specify the object invoked for 'createElement'. This only applies when targeting 'react' JSX emit. */
+        // "noLib": true,                                    /* Disable including any library files, including the default lib.d.ts. */
+        // "useDefineForClassFields": true,                  /* Emit ECMAScript-standard-compliant class fields. */
+        // "moduleDetection": "auto",                        /* Control what method is used to detect module-format JS files. */
+        /* Modules */
+        "module": "ES2020", /* Specify what module code is generated. */
+        "rootDir": "./", /* Specify the root folder within your source files. */
+        "moduleResolution": "node", /* Specify how TypeScript looks up a file from a given module specifier. */
+        // "baseUrl": ".",                                   /* Specify the base directory to resolve non-relative module names. */
+        // "paths": {},                                      /* Specify a set of entries that re-map imports to additional lookup locations. */
```

I would add a path here for shared to make imports better readable
```suggestion
         "paths": {
             "@shared/*": ["src/shared/*"], /* Specify a set of entries that re-map imports to additional lookup locations. */
         },
```

---

I also would like this feature, but we are using esm modules. The typescript compiler will not rewrite the import and export statements. The tsconfig-paths plugin only intercepts import calls from the commonjs loader during runtime and has no affect when using esm. Currently there is no out of the box way which will work with esm.